handler/src/main/java/io/netty/handler/ssl/AbstractSniHandler.java

```diff
@@ -250,6 +252,15 @@ public void operationComplete(Future<T> future) throws Exception {
         }
     }
 
+    private void fireSniCompletionEvent(ChannelHandlerContext ctx, String hostname, Future<T> future) {
+        Throwable cause = future.cause();
+        if (cause == null) {
+            ctx.fireUserEventTriggered(new SniCompletionEvent(hostname));
+        } else {
+            ctx.fireUserEventTriggered(new SniCompletionEvent(cause));
```

Comment:
I think this should be `new SniCompletionEvent(hostname, cause)` as it's possible to have a failed future in conjunction with a hostname.

Reply:
Yes makes sense